### PR TITLE
chore: update guild feature list

### PR DESCRIPTION
**Pull request by onerandomusername**
## Summary
adds a few missing features and marks a few others as no more effect.

## Checklist

- [ ] If code changes were made, then they have been tested
    - [ ] I have updated the documentation to reflect the changes
    - [ ] I have formatted the code properly by running `task lint`
    - [ ] I have type-checked the code by running `task pyright`
- [ ] This PR fixes an issue
- [ ] This PR adds something new (e.g. new method or parameters)
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

File: disnake/guild.py [hunk 1]
```diff
@@ -222,6 +222,7 @@ class Guild(Hashable):
         - ``HUB``: Guild is a student hub.
         - ``INVITE_SPLASH``: Guild's invite page can have a special splash.
         - ``LINKED_TO_HUB``: Guild is linked to a student hub.
+        - ``MEMBER_PROFILES``: Guild members can set per-guild profiles if they have nitro.
```
Comment (shiftinv) [inline disnake/guild.py:225]: this isn't documented, and doesn't seem to do anything

**Review by shiftinv — COMMENTED** [review]
`THREADS_ENABLED`, `NEW_THREAD_PERMISSIONS` and `TEXT_IN_VOICE_ENABLED` also no longer have an effect

File: disnake/guild.py (hunk 1 above)
Comment (onerandomusername) [reply disnake/guild.py]: https://github.com/DisnakeDev/disnake/pull/694/commits/8d2adfa6b5c8381bc9bf1bad4638d9fd1134d409

**Comment by onerandomusername** [comment]
Still need to check on this list a bit more, I think a few others have been removed/deprecated on this list.

**Comment by onerandomusername** [comment]
Added a few currently undocumented features to types/guild but did not document them in our own documentation.

**Review by shiftinv — APPROVED** [review]
lgtm

**Pull request merged**